src/store/poolSlice.ts

```diff
@@ -0,0 +1,444 @@
+    args: Omit<LPSupplyWithPermitType, 'user'>
+  ) => Promise<EthereumTransactionTypeExtended[]>;
+  setUserEMode: (categoryId: number) => Promise<EthereumTransactionTypeExtended[]>;
+  signERC20Approval: (args: Omit<LPSignERC20ApprovalType, 'user'>) => Promise<string>;
+  claimRewards: (args: ClaimRewardsActionsProps) => Promise<EthereumTransactionTypeExtended[]>;
+  // TODO: optimize types to use only neccessary properties
+  swapCollateral: (args: SwapActionProps) => Promise<EthereumTransactionTypeExtended[]>;
+  repay: (args: RepayActionProps) => Promise<EthereumTransactionTypeExtended[]>;
+  repayWithPermit: (
+    args: RepayActionProps & {
+      signature: SignatureLike;
+      deadline: string;
+    }
+  ) => Promise<EthereumTransactionTypeExtended[]>;
+  supply: (
+    args: Omit<SupplyActionProps, 'poolReserve'>
+  ) => Promise<EthereumTransactionTypeExtended[]>;
+}
+
+export const createPoolSlice: StateCreator<
+  RootStore,
+  [['zustand/devtools', never]],
+  [],
+  PoolSlice
+> = (set, get) => {
+  function getCorrectPool() {
+    const currentMarketData = get().currentMarketData;
+    const provider = get().jsonRpcProvider();
+    if (currentMarketData.v3) {
+      return new Pool(provider, {
+        POOL: currentMarketData.addresses.LENDING_POOL,
+        REPAY_WITH_COLLATERAL_ADAPTER: currentMarketData.addresses.REPAY_WITH_COLLATERAL_ADAPTER,
+        SWAP_COLLATERAL_ADAPTER: currentMarketData.addresses.SWAP_COLLATERAL_ADAPTER,
+        WETH_GATEWAY: currentMarketData.addresses.WETH_GATEWAY,
+        L2_ENCODER: currentMarketData.addresses.L2_ENCODER,
+      });
+    } else {
+      return new LendingPool(provider, {
+        LENDING_POOL: currentMarketData.addresses.LENDING_POOL,
+        REPAY_WITH_COLLATERAL_ADAPTER: currentMarketData.addresses.REPAY_WITH_COLLATERAL_ADAPTER,
+        SWAP_COLLATERAL_ADAPTER: currentMarketData.addresses.SWAP_COLLATERAL_ADAPTER,
+        WETH_GATEWAY: currentMarketData.addresses.WETH_GATEWAY,
+      });
+    }
+  }
+  return {
+    data: new Map(),
+    refreshPoolData: async () => {
+      const account = get().account;
+      const currentMarketData = get().currentMarketData;
+      const currentChainId = get().currentChainId;
+      const poolDataProviderContract = new UiPoolDataProvider({
+        uiPoolDataProviderAddress: currentMarketData.addresses.UI_POOL_DATA_PROVIDER,
+        provider: get().jsonRpcProvider(),
+        chainId: currentChainId,
+      });
+      const lendingPoolAddressProvider = currentMarketData.addresses.LENDING_POOL_ADDRESS_PROVIDER;
+      const promises: Promise<void>[] = [];
+      try {
+        promises.push(
+          poolDataProviderContract
+            .getReservesHumanized({
+              lendingPoolAddressProvider,
+            })
+            .then((reservesResponse) =>
+              set((state) =>
+                produce(state, (draft) => {
+                  if (!draft.data.get(currentChainId)) draft.data.set(currentChainId, new Map());
```

Comment:
I think it'd be good to fix the non-null assertions, so we're not bypassing our `strict` compiler options.

I noticed our linter config doesn't pick this up as warning, it's an issue with next js lint directory config. We'll fix that separately, but if you run `next lint --dir src` you should see those warnings get flagged.